packages/react/src/analytics/integrations/Vitorino/providers/Riskified/Riskified.ts

```diff
@@ -0,0 +1,149 @@
+import get from 'lodash/get';
+import type { RiskifiedIntegrationOptions } from './types';
+
+class Riskified extends integrations.Integration<RiskifiedIntegrationOptions> {
+  private isReady: boolean;
+  private siteId: string;
+  private baseUrl: string;
+
+  /**
+   * Creates an instance of Riskified and sets the script on the page.
+   *
+   * @param options - Custom options for the integration.
+   * @param loadData - Analytics's load event data.
+   */
+  constructor(
+    options: RiskifiedIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, analytics);
+
+    this.isReady = false;
+    this.siteId = options.siteId || 'farfetch.com';
+    this.baseUrl = `https://beacon.riskified.com?shop=${this.siteId}&sid=`;
+
+    this.setupScript(loadData);
+  }
+
+  /* static createInstance<Options extends IntegrationOptions>(
+    this: IntegrationFactory<Options>,
+    options: Options,
+*/
+  static override createInstance<Options extends RiskifiedIntegrationOptions>(
+    this: IntegrationFactory<Options>,
+    options: Options,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    return new this(options, loadData, analytics);
+  }
+
+  /**
+   * Returns true due to being a required integration - No need to check for consent.
+   *
+   * @returns A value indicating if the integration should load.
+   */
+  static override shouldLoad(): boolean {
+    return true;
+  }
+
+  /**
+   * Creates the script based on the correlation `id` and appends it on the page.
+   * Once the script is on the page, it will automatically call `RISKX.go()`.
+   *
+   * @param loadData - Analytics's load event data.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  setupScript(loadData: LoadIntegrationEventData): this {
+    const correlationId = get(loadData, 'user.localId');
+    const script = document.createElement('script');
+    const url = `${this.baseUrl}${correlationId}`;
+
+    document.body.appendChild(script);
+
+    script.onload = this.onLoad;
+
+    script.setAttribute('data-test', 'riskified');
+    script.async = true;
+    script.src = url;
+
+    return this;
+  }
+
+  /**
+   * Tracks events (page views or page actions).
+   *
+   * @param data - The event object.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  override track(data: EventData<TrackTypesValues>) {
+    switch (data.type) {
+      case TrackTypes.PAGE:
+        return this.trackPage(get(data, 'context.web.window.location.href'));
+
+      case TrackTypes.TRACK:
+        return null;
+
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Tracks the page view event. Calls the global Riskified variable RISKX.go with the web location.
+   *
+   * @param location - The location data.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  trackPage(location: string): this {
```

Comment:
Track page is returning the instance but afaics it is not being used anywhere. This method is being called by the track method which is called by analytics which does not do anything with the return value.

Reply:
Removed the return in this method.